plugins/inputs/ssl_cert/ssl_cert.go

```diff
@@ -0,0 +1,164 @@
+package ssl_cert
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const sampleConfig = `
+  ## List of local SSL files
+  # files = []
+  ## List of servers
+  # servers = []
+  ## Timeout for SSL connection
+  # timeout = 5
+`
+const description = "Reads metrics from a SSL certificate"
+
+// SSLCert holds the configuration of the plugin.
+type SSLCert struct {
+	Servers []string      `toml:"servers"`
+	Files   []string      `toml:"files"`
+	Timeout time.Duration `toml:"timeout"`
+
+	// For tests
+	CloseConn  bool
```

Comment:
As I said in https://github.com/influxdata/telegraf/pull/3768#discussion_r166923553 I would prefer to see black box tests where your tests live in separate package `ssl_cert_test`, and therefore you test only the public API. But if you have to have access to parts of the package, which change the behaviour, so that more test cases can be covered, then I would recommend the following:

Move the test-specific code outside the configuration struct `SSLCert`

```go
// SSLCert holds the configuration of the plugin.
type SSLCert struct {
	Servers []string      `toml:"servers"`
	Files   []string      `toml:"files"`
	Timeout time.Duration `toml:"timeout"`
}

// For tests
var (
	closeConn  bool
	unsetCerts bool
)
```

This way the test-related code does not become part of the public API.

The disadvantage of having global state modifiers for the package by using this approach is that you will not be able to run tests in parallel, which you do not do here anyway.

Reply:
Thanks for your review. It should be fixed in the last commit.